<launch>
    <group ns="webcam">
        <node pkg="libuvc_camera" type="camera_node" name="c920">
            <!-- Parameters used to find the camera -->
            <param name="vendor" value="0x046d"/>
            <param name="product" value="0x082d"/>
            <param name="serial" value=""/>
            <!-- If the above parameters aren't unique, choose the first match: -->
            <param name="index" value="0"/>

            <!-- Image size and type -->
            <param name="width" value="160"/>
            <param name="height" value="90"/>
            <!-- choose whichever uncompressed format the camera supports: -->
            <param name="video_mode" value="uncompressed"/> <!-- or uncompressed/yuyv/nv12/mjpeg -->
            <param name="frame_rate" value="30"/>

            <param name="timestamp_method" value="start"/> <!-- start of frame -->
            <param name="camera_info_url" value="file:///tmp/cam.yaml"/>

            <param name="auto_exposure" value="0"/>
            <!-- <param name="exposure_absolute" value="0.001"/> -->

            <param name="auto_white_balance" value="false"/>
            <param name="auto_focus" value="true"/>
        </node>
    </group>
    <group ns="pelicannon">
        <node pkg="pelicannon" type="node_k66f.py" name="k66f" output="screen">
            <param name="run_calibration" value="false"/>
            <param name="gyro_cal_x" value="-0.011587"/>
            <param name="gyro_cal_y" value="0.019745"/>
            <param name="gyro_cal_z" value="0.002371"/>
            <param name="tty" value="/dev/ttyTHS1"/>
            <param name="baud" value="115200"/>
        </node>
        <node pkg="pelicannon" type="node_object_detector.py" name="object_detector" output="screen">
            <param name="motion_regions" value="true"/>
            <param name="body_regions" value="true"/>
            <param name="debug" value="true"/>
        </node>
        <node pkg="pelicannon" type="node_debug.py" name="debug" output="screen">
            <!-- Options include /pelicannon/image_transform, /pelicannon/image_abs_diff, /pelicannon/image_thresh, and /webcam/image_raw -->
            <param name="video_source" value="/pelicannon/image_raw"/>
        </node>
        <node pkg="pelicannon" type="node_base.py" name="base" output="screen">
            <param name="camera_height" value="1.0"/>
        </node>
    </group>
    <node pkg="imu_filter_madgwick" type="imu_filter_node" name="madgwick" output="screen">
        <param name="use_mag" value="true"/>
        <param name="use_magnetic_field_msg" value="true"/>
        <param name="stateless" value="false"/>
        <param name="constant_dt" value="0.0"/>
        <param name="gain" value="0.5"/>
    </node>

</launch>
